### PR TITLE
WIP: Add github workflow for CI

### PR DESCRIPTION
**Pull request by justinsb**
The equivalent of the travis workflow.

### CODE REVIEW

**Comment by aojea** [comment]
it seems that there is no CI gating in this repo, is there @lavalamp ?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  GOPROXY: https://proxy.golang.org
+
+jobs:
+  verify:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.1
```
Comment (aojea) [inline .github/workflows/main.yml:18]:
```suggestion
    strategy:
      fail-fast: false
      matrix:
        go-version: [1.18.x, 1.19.x]
    runs-on: ubuntu-20.04
    steps:
      - name: Set up go
        uses: actions/setup-go@v3
        with:
          go-version: ${{ matrix.go-version }}
```

**Comment by aojea** [comment]
/assign @thockin @lavalamp 

I think this repo is merging without running the tests first :thinking:

**Comment by lavalamp** [comment]
heh this explains some things, sorry I didn't see this when it was sent.

/approve

Comment looks worth addressing? (I don't know how this file works)

**Comment by thockin** [comment]
Are we using GH actions for "small" repos or would it be better to teach Prow?

**Comment by aojea** [comment]
My experience for small repos is that actions is faster and simpler

**Comment by aojea** [comment]
> Are we using GH actions for "small" repos or would it be better to teach Prow?

we can do both too https://github.com/kubernetes/test-infra/pull/27586

**Comment by aojea** [comment]
/unassign @wojtek-t @ameukam @thockin @lavalamp

ups, wrong tab sorry

**Comment by aojea** [comment]
/close

Prow config has merged and is working, I think there is no need to duplicate efforts

https://github.com/kubernetes/test-infra/pull/27586

**Pull request closed without merging**